Cannot find module 'path/win32' with Shescape 1.5.3

# Bug Report

- **Shescape version**: `1.5.3`
- **Operating system**: `ubuntu-20.04`
- **Shell**: n/a

## Description

On non-Windows systems there's a change that an error is thrown saying `Cannot find module 'path/win32'`.

### Actual Behaviour

The following error is thrown during loading.

```
internal/modules/cjs/loader.js:818
  throw err;
  ^

Error: Cannot find module 'path/win32'
Require stack:
- /home/runner/work/git-tag-annotation-action/git-tag-annotation-action/lib/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/runner/work/git-tag-annotation-action/git-tag-annotation-action/lib/index.js:1:251)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/runner/work/git-tag-annotation-action/git-tag-annotation-action/lib/index.js'
  ]
}
```

### Expected Behaviour

No error should be thrown due to the use of `path/win32` on non-Windows systems.

## Working Example

See: https://github.com/ericcornelissen/git-tag-annotation-action/runs/5444956321?check_suite_focus=true#step:6:1

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -32,14 +32,13 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest, windows-latest]
-        node-version: [10.13.0, 12, 14, 16, 17]
     steps:
       - name: Check out code
         uses: actions/checkout@v3.0.0
       - name: Install Node.js
         uses: actions/setup-node@v3.0.0
         with:
-          node-version: ${{ matrix.node-version }}
+          node-version-file: .nvmrc
           cache: npm
       - name: Install dependencies
         run: npm ci
@@ -209,13 +208,17 @@ jobs:
   transpile:
     name: CommonJS
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        node-version: [10.13.0, 12, 14, 16, 17]
     steps:
       - name: Check out code
         uses: actions/checkout@v3.0.0
       - name: Install Node.js
         uses: actions/setup-node@v3.0.0
         with:
-          node-version-file: .nvmrc
+          node-version: ${{ matrix.node-version }}
           cache: npm
       - name: Install dependencies
         run: npm ci

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -32,13 +32,14 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest, windows-latest]
+        node-version: [10.13.0, 12, 14, 16, 17]
     steps:
       - name: Check out code
         uses: actions/checkout@v3.0.0
       - name: Install Node.js
         uses: actions/setup-node@v3.0.0
         with:
-          node-version-file: .nvmrc
+          node-version: ${{ matrix.node-version }}
           cache: npm
       - name: Install dependencies
         run: npm ci

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,7 +7,7 @@ Versioning].
 
 ## [Unreleased]
 
-- _No changes yet_
+- Fix `Cannot find module 'path/win32'` error on Node.js <16.0.0. ([#199])
 
 ## [1.5.3] - 2022-03-06
 
@@ -122,5 +122,6 @@ Versioning].
 [#173]: https://github.com/ericcornelissen/shescape/pull/173
 [#178]: https://github.com/ericcornelissen/shescape/pull/178
 [#190]: https://github.com/ericcornelissen/shescape/pull/190
+[#199]: https://github.com/ericcornelissen/shescape/pull/199
 [keep a changelog]: https://keepachangelog.com/en/1.0.0/
 [semantic versioning]: https://semver.org/spec/v2.0.0.html

--- a/src/win.js
+++ b/src/win.js
@@ -5,7 +5,7 @@
  */
 
 import * as fs from "fs";
-import * as path from "path/win32";
+import * as path from "path";
 import which from "which";
 
 /**
@@ -102,7 +102,7 @@ const quoteFunctionsByShell = new Map([
  * @returns {string} The basename of `fullPath`.
  */
 function getBasename(fullPath) {
-  return path.basename(fullPath);
+  return path.win32.basename(fullPath);
 }
 
 /**

--- a/test/index.test.cjs
+++ b/test/index.test.cjs
@@ -16,7 +16,7 @@ describe("index.cjs", function () {
     const input = "Hello world!";
     const result = shescape.escape(input);
     assert.notStrictEqual(result, "");
-    assert.doesNotMatch(result, /^(?<q>"|').*(\k<q>)$/);
+    assert.ok(!/^(?<q>"|').*(\k<q>)$/.test(result));
   });
 
   it("has a functioning `escapeAll` function", function () {
@@ -27,7 +27,7 @@ describe("index.cjs", function () {
     const result = shescape.escapeAll(inputs);
     for (const output of result) {
       assert.notStrictEqual(output, "");
-      assert.doesNotMatch(output, /^(?<q>"|').*(\k<q>)$/);
+      assert.ok(!/^(?<q>"|').*(\k<q>)$/.test(output));
     }
   });
 
@@ -38,7 +38,7 @@ describe("index.cjs", function () {
     const input = "Hello world!";
     const result = shescape.quote(input);
     assert.notStrictEqual(result, "");
-    assert.match(result, /^(?<q>"|').*(\k<q>)$/);
+    assert.ok(/^(?<q>"|').*(\k<q>)$/.test(result));
   });
 
   it("has a functioning `quoteAll` function", function () {
@@ -49,7 +49,7 @@ describe("index.cjs", function () {
     const result = shescape.quoteAll(inputs);
     for (const output of result) {
       assert.notStrictEqual(output, "");
-      assert.match(output, /^(?<q>"|').*(\k<q>)$/);
+      assert.ok(/^(?<q>"|').*(\k<q>)$/.test(output));
     }
   });
 });

